data_structures/trie/trie.py

```diff
@@ -56,20 +85,25 @@ def print_words(node: TrieNode, word: str):  # noqa: E999 This syntax is Python
     :return: None
     """
     if node.is_leaf:
-        print(word, end=' ')
+        print(word, end=" ")
 
     for key, value in node.nodes.items():
         print_words(value, word + key)
 
 
 def test():
-    words = ['banana', 'bananas', 'bandana', 'band', 'apple', 'all', 'beast']
+    words = ["banana", "bananas", "bandana", "band", "apple", "all", "beast"]
     root = TrieNode()
     root.insert_many(words)
     # print_words(root, '')
-    assert root.find('banana')
-    assert not root.find('bandanas')
-    assert not root.find('apps')
-    assert root.find('apple')
+    assert root.find("banana")
+    assert not root.find("bandanas")
+    assert not root.find("apps")
+    assert root.find("apple")
+    root.delete("all")
+    assert not root.find("all")
+    root.delete("banana")
+    assert not root.find("banana")
```

Comment:
Just below this line, please add a __assert root.find("banana")__ to prove that removing the singular does not remove the plural.